File: Leetcode-Problems/Missing Number.cpp
class Solution
{
public:
    int missingNumber(vector<int> &nums)
    {
        sort(nums.begin(), nums.end());
        int ans = 0;
        for (int i : nums)
        {
            if (i != ans)
            {
                break;
            }
            ans++;
        }
        return ans;
    }
};

File: Leetcode-Problems/Missing Number.java
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int x = 0, y = 0;
        for (int i = 0; i <= n; i++) {
            x ^= i;
        }
        for (int i : nums) {
            y ^= i;
        }
        return x ^ y;
    }
}